compiler/rustc_middle/src/mir/interpret/mod.rs

```diff
@@ -671,11 +671,11 @@ pub fn read_target_uint(endianness: Endian, mut source: &[u8]) -> Result<u128, i
     // So we do not read exactly 16 bytes into the u128, just the "payload".
     let uint = match endianness {
         Endian::Little => {
-            source.read(&mut buf)?;
+            let _ = source.read(&mut buf)?;
```

Comment:
read_exact here, too?

Reply:
I already tried that and got failures in CI https://github.com/rust-lang/rust/actions/runs/8307751008/job/22737303335#step:27:1679. And that actually makes sense as trying to fill entire `[0u8; std::mem::size_of::<u128>()]` will fail for most cases.

Reply:
right, you need to use the same `[16 - source.len()..]` logic used in the `Endian::Big` branch (or even better, move that logic before the `match endianess`

Reply:
I believe this is to cover the logic for the "most significant byte" and we shouldn't do the same in `Little` branch.

Reply:
Ah makes sense. Then you need to slice `..source.len()` before using `read_exact`

Tho maybe we should just use one of the slice copying functions instead of using the Read trait